backend/routes/teamspace.js

```diff
@@ -368,6 +593,102 @@
 			});
 	}
 
+	function getTeamspaceSettings(req, res, next) {
+		TeamspaceSettings.getTeamspaceSettings(req.params.account).then((settings) => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, settings);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function updateTeamspaceSettings(req, res, next) {
+		TeamspaceSettings.update(req.params.account, req.body).then((settings) => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, settings);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function getMitigationsFile(req, res, next) {
+		FileRef.getMitigationsFile(req.params.account).then((mitigations) => {
+			const timestamp = (new Date()).toLocaleString();
+			const filenamePrefix = (req.params.account + "_" + timestamp + "_").replace(/\W+/g, "_");
+
+			const headers = {
+				"Content-Disposition": "attachment;filename=" + filenamePrefix + "mitigations.csv",
+				"Content-Type": "text/csv"
+			};
+
+			res.set(headers);
+
+			responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, mitigations);
+		}).catch(err => {
+			responseCodes.respond(utils.APIInfo(req), req, res, next, err, err);
+		});
+	}
+
+	function uploadMitigationsFile(req, res, next) {
+		const place = utils.APIInfo(req);
+		const {account} = req.params;
+		const sessionId = req.headers[C.HEADER_SOCKET_ID];
+		const user = req.session.user.username;
+
+		function fileFilter(fileReq, file, cb) {
+			const acceptedFormat = ["csv"];
+
+			let format = file.originalname.split(".");
+			format = format.length <= 1 ? "" : format.splice(-1)[0];
+
+			const size = parseInt(fileReq.headers["content-length"]);
+
+			if (acceptedFormat.indexOf(format.toLowerCase()) === -1) {
+				return cb({ resCode: responseCodes.FILE_FORMAT_NOT_SUPPORTED });
+			}
+
+			if (size > config.uploadSizeLimit) {
+				return cb({ resCode: responseCodes.SIZE_LIMIT });
+			}
+
+			cb(null, true);
+		}
+
+		if (!config.bcf_dir) {
```

Comment:
I don't think bcf_dir is relevant here? just store it in the uploads directory like the models. i don't know why we need more than the uploads directory really, bcf should really go there too